assert/assertions.go

```diff
@@ -1343,9 +1343,15 @@ func PanicsWithError(t TestingT, errString string, f PanicTestFunc, msgAndArgs .
 	if !funcDidPanic {
 		return Fail(t, fmt.Sprintf("func %#v should panic\n\tPanic value:\t%#v", f, panicValue), msgAndArgs...)
 	}
-	panicErr, ok := panicValue.(error)
-	if !ok || panicErr.Error() != errString {
-		return Fail(t, fmt.Sprintf("func %#v should panic with error message:\t%#v\n\tPanic value:\t%#v\n\tPanic stack:\t%s", f, errString, panicValue, panickedStack), msgAndArgs...)
+	panicErr, isError := panicValue.(error)
+	if !isError || panicErr.Error() != errString {
+		msg := fmt.Sprintf("func %#v should panic with error message:\t%#v\n", f, errString)
+		if isError {
+			msg += fmt.Sprintf("\tError message:\t%#v\n", panicErr.Error())
+		}
+		msg += fmt.Sprintf("\tPanic value:\t%#v\n", panicValue)
```

Comment:
Is there a need for panic value when isError is satisfied?


```suggestion
			msg += fmt.Sprintf("\tError message:\t%#v\n", panicErr.Error())
		} else {
			msg += fmt.Sprintf("\tPanic value:\t%#v\n", panicValue)
		}
```

Reply:
Indeed.

Maybe tiu should add tests about this, an error providing more info. I'm unsure how it can happen.